SerialPrograms/Source/PokemonSV/Programs/ItemPrinter/PokemonSV_ItemPrinterRNG.cpp

```diff
@@ -697,6 +705,377 @@ void ItemPrinterRNG::run_print_at_date(
         }
     }
 }
+
+void ItemPrinterRNG::adjust_delay(
+    std::array<std::string, 10> print_results, 
+    uint64_t seed, ConsoleHandle& console
+) {
+    DistanceFromTarget distance_from_target = get_distance_from_target(print_results, seed, console);
+    int16_t delay_adjustment;
+    int16_t current_delay_mills = DELAY_MILLIS;
+    switch (distance_from_target){
+    case DistanceFromTarget::UNKNOWN:
+    case DistanceFromTarget::ON_TARGET:
+        delay_adjustment = 0;
+        break;
+    case DistanceFromTarget::MINUS_1:
+        delay_adjustment = -50;
+        break;
+    case DistanceFromTarget::MINUS_2:
+        delay_adjustment =  -1000;
+        break;
+    case DistanceFromTarget::PLUS_1:
+        delay_adjustment =  50;
+        break;
+    case DistanceFromTarget::PLUS_2:
+        delay_adjustment = 1000;
+        break;        
+    default:
+        delay_adjustment = 0;
+        break;
+    }
+
+    int16_t new_delay = current_delay_mills + delay_adjustment;
+    if (new_delay < 0){
+        new_delay = 0;
+    }
+    DELAY_MILLIS.set((uint16_t) new_delay);
+    console.log("Current delay: " + std::to_string(new_delay));
+    // std::cout << "Current delay:" << new_delay << std::endl;
+    
+}
+
+DistanceFromTarget ItemPrinterRNG::get_distance_from_target(
+    std::array<std::string, 10> print_results, 

```

Comment:
Always take large objects by reference if they don't need to be copied. (And const-reference if they don't need to be modified)